### PR TITLE
Update Taxon specification.html

### PR DESCRIPTION
**Pull request by frmichel**
Hi Ricardo, 

This is my first time updating the specification, not sure I'm doing this right. I pasted the YAML file generated by Goweb into this one, only replacing the spec_info and mapping sections.

Let me know if this is ok.

Franck.

### CODE REVIEW

**Comment by frmichel** [comment]
Hey guys, could you please review this PR? I've started inviting people from the biodiversity community. I've already updated the example but I'd prefer to have the specification in line when discussions start.
Thx,
   Franck.

**Comment by ricardoaat** [comment]
Hi Frank!

I have tested locally and everything seems to be in order. This is ready to be merge.

Kind regards,
Ricardo

**Comment by frmichel** [comment]
Ok great; thx @ricardoaat. 
Now what is the process for the [specification web page](http://bioschemas.org/devSpecs/Taxon/specification/) to be updated with the changes?

**Comment by ricardoaat** [comment]
@frmichel it will be automatically updated as soon as it get merged to master. It would be great to have another approval for this PR (2 its the sweet spot), @AlasdairGray, @ljgarcia or @rajido could you help us with one quick review? The changes are small, so they would not break anything. I've tested locally just in case.

**Comment by frmichel** [comment]
@AlasdairGray, @ljgarcia or @rajido, sorry to get on that again ;). 
Could any of you approve the PR so that we can merge it?
Thx in advance,
   Franck.

**Comment by frmichel** [comment]
@ricardoaat, the PR now shows a conflict on _devSpecs/Taxon/specification.html. How come?

**Comment by ricardoaat** [comment]
@frmichel Seems like on a recent commit a new property were added https://github.com/BioSchemas/bioschemas.github.io/commit/4b921cd079b08da8cfd0f284317fa745d82a477b#diff-ca5160a1fc59bfd6a9b906cd3e41c697 all you have to do to resolve that conflict is to add that property I think, @ljgarcia isn't this right?

**Comment by AlasdairGray** [comment]
> @AlasdairGray, @ljgarcia or @rajido, sorry to get on that again ;).
> Could any of you approve the PR so that we can merge it?

Sorry, last week got away from me. I'm looking at it now.

File: _devSpecs/Taxon/specification.html [hunk 1]
```diff
@@ -78,29 +77,36 @@
...
+        "name": "Delphinapterus leucas",
+        "dwc:scientificName": "Delphinapterus leucas (Pallas, 1776)",
+        "dwc:scientificNameAuthorship": "(Pallas, 1776)"
     }
 - property: rank
   expected_types:
   - Text
   description: ""
   type: external
   type_url: http://rs.tdwg.org/ontology/voc/TaxonConcept#rank
```
[the remark this answers is not in the document]
Comment (frmichel) [reply _devSpecs/Taxon/specification.html]: > you use the property scientificName as if it is part of schema.org, but currently it does not exist. 

This is not my intend, I may simply have misused the specification spreadsheet: I have used the type_url field to give the property url (when it comes from another ontology), but I realise I did not use the controlled_vocab field:

```
- property: scientificName
  type_url: http://rs.tdwg.org/dwc/terms/scientificName
  marginality: Optional
  cardinality: ONE
  controlled_vocab: ""
```

Can you clarify the way I should do that?

> The url for rank does not resolve.

Right. Actually the URIs of the TDWG's ontologies are not dereferenceable. Is it a show stopper? I mean, should we NOT use such URIs?

> What is the difference between rank and taxonRank?

tc:rank is meant to be used with URIs, while dwc:taxonRank is meant to be used with string literals. Ideally we should obviously use URIs from a given vocabulary. However in the biodiv community, many resources so far have used literals. So this is basically a pragmatic approach to enable the publication even if the only thing we have are literals.

**Comment by ljgarcia** [comment]
> 
> 
> @ricardoaat, the PR now shows a conflict on _devSpecs/Taxon/specification.html. How come?

Hi @frmichel I updated your branch from the master, no conflict detected

**Comment by frmichel** [comment]
Thx @ljgarcia.

As @AlasdairGray pointed out, there may be a few issues, related to me misusing the specification spreadsheet. I copy and paste the explanations below.


> you use the property scientificName as if it is part of schema.org, but currently it does not exist.

This was not my intend, I may simply have misused the specification spreadsheet: I have used the type_url field to give the property url (when it comes from another ontology), but I realise I did not use the controlled_vocab field:
 
 ```
 - property: scientificName
   type_url: http://rs.tdwg.org/dwc/terms/scientificName
   marginality: Optional
   cardinality: ONE
   controlled_vocab: ""
 ```
 
 Can you clarify the way I should do that?
 
 > The url for rank does not resolve.
 
 Right. Actually the URIs of the TDWG's ontologies are not dereferenceable. Is it a show stopper? I mean, should we NOT use such URIs?
 
 > What is the difference between rank and taxonRank?
 
 tc:rank is meant to be used with URIs, while dwc:taxonRank is meant to be used with string literals. Ideally we should obviously use URIs from a given vocabulary. However in the biodiv community, many resources so far have used literals. So this is basically a pragmatic approach to enable the publication even if the only thing we have are literals.
Perhaps it would be preferable to say the tc:rank is recommended and dwc:taxonRank is optional?

File: _devSpecs/Taxon/specification.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ljgarcia) [reply _devSpecs/Taxon/specification.html]: Hi @frmichel

Regarding the rank and the taxonRank, maybe you could use the property **hasCategoryCode**, see http://bioschemas.org/devTypes/BioChemEntity/. In your case, the **CategoryCode** would have the **codeValue** pointing to what you currently have under **rank**, the **inCodeset** would point to the vocabulary where the **rank** term comes, and you still can use **name** for the **taxonRank**. Would that work for you?

@AlasdairGray I will leave with you the pending reply regarding dereferenceable URIs

@ricardoaat could you please double check the mapping file is correct regarding terms coming from external ontologies?

Cheers!

[the remark this answers is not in the document]
Comment (ljgarcia) [reply _devSpecs/Taxon/specification.html]: Oh! Forgot about the scientificName. @frmichel Please have a look at http://bioschemas.org/devSpecs/Protein/, particularly the property **associatedWith**. That one comes from an external ontology, as your scientificName does. I think it is alright in your spec. @AlasdairGray any further comments on that one?

**Comment by frmichel** [comment]
Note that I've deleted my recent comment where I copied/paste an [ealier comment](https://github.com/BioSchemas/bioschemas.github.io/pull/132#discussion_r219838842) to which @ljgarcia answered a few minutes ago. This is just to avoid mixing things up.

**Comment by AlasdairGray** [comment]
@frmichel I see that you've moved the properties around a bit now. That is fine. However, the text for rank does not correspond to the type; descriptive text states that it should be a URI while the expected type is declared to be text.

**Comment by ljgarcia** [comment]
Not sure this comment is still needed, but just in case...

Hi @frmichel

Regarding the rank and the taxonRank, maybe you could use the property hasCategoryCode, see http://bioschemas.org/devTypes/BioChemEntity/. In your case, the CategoryCode would have the codeValue pointing to what you currently have under rank, the inCodeset would point to the vocabulary where the rank term comes, and you still can use name for the taxonRank. Would that work for you?

@ricardoaat could you please double check the mapping file is correct regarding terms coming from external ontologies?

Oh! Forgot about the scientificName. @frmichel Please have a look at http://bioschemas.org/devSpecs/Protein/, particularly the property associatedWith. That one comes from an external ontology, as your scientificName does. I think it is alright in your spec. @AlasdairGray any further comments on that one?

**Comment by AlasdairGray** [comment]
As with the Protein [spec](http://bioschemas.org/specifications/Protein/), can we indicate in the name of a property from an external ontology which ontology it is taken from. Adding a prefix of `tc` and `dwc` would be fine

**Comment by AlasdairGray** [comment]
Examples don't correspond with the property and the cardinality, e.g. `name` has three examples only one of which is the name, `rank` has a declared cardinality of one but then has multiple values.

**Comment by frmichel** [comment]
Hi @ljgarcia, @AlasdairGray, thx for all these remarks. I'm fixing them in the [spreadsheet](https://docs.google.com/spreadsheets/d/1hHE7SHz9qt6Eg6j6DQgI9BwhrlGOwkO68fzek78m62I/edit#gid=1261485211) and will regenerate the yaml file once this is ok.

> the text for rank does not correspond to the type; descriptive text states that it should be a URI while the expected type is declared to be text.
> rank has a declared cardinality of one but then has multiple values.

Fixed.

> Examples don't correspond with the property e.g. name has three examples only one of which is the name, 

Right, my intend was to show the usage in context, exemplifying the relationship between name, scientificName and sicnetificNameAuthorhip. Fixed.

> As with the Protein spec, can we indicate in the name of a property from an external ontology which ontology it is taken from. Adding a prefix of tc and dwc would be fine

Ok, that's a good visual indicator indeed, it complements the html anchor that will be generated. Note that this applies in Protein current spec (v4) but not in the devspec (v5), see for instance enableMF or associatedWith.

> Regarding the rank and the taxonRank, maybe you could use the property hasCategoryCode, see http://bioschemas.org/devTypes/BioChemEntity/. In your case, the CategoryCode would have the codeValue pointing to what you currently have under rank, the inCodeset would point to the vocabulary where the rank term comes, and you still can use name for the taxonRank. Would that work for you?

I had not seen that but that's interesting indeed. However, I'd suggest to keep this for a later discussion with other biodiversity people. Is it ok with you @ljgarcia?

>>>  The url for rank does not resolve.
>>
>> Right. Actually the URIs of the TDWG's ontologies are not dereferenceable. Is it a show stopper? I mean, should we NOT use such URIs?
>
> No, this is not a show stopper, although it is desirable that they are dereferencable. It just makes it harder to verify the property has the correct IRI.

I agree. Actually I was willing to keep this for a later discussion. The current specification status is really just a first draft, but the choice of appropriate vocabularies needs to be discussed with the community.

**Comment by ljgarcia** [comment]
> 
> 
> As with the Protein [spec](http://bioschemas.org/specifications/Protein/), can we indicate in the name of a property from an external ontology which ontology it is taken from. Adding a prefix of `tc` and `dwc` would be fine

@frmichel No, please do not add any prefixes.

@AlasdairGray all those prefixes will go away with the context. The original property source will only be visible inside the jsonld automatically created fromt the YAML. Properties from external ontologies are also used in schema.org, see for instance ScholarlyArticle where pageStart is equivalent to bibo:pageStart, that is not visible on the html but on the jsonld

```
 {
      "@id": "schema:pageStart",
      "@type": "rdf:Property",
      "dct:source": {
        "@id": "http://www.w3.org/wiki/WebSchemas/SchemaDotOrgSources#source_bibex"
      },
      "owl:equivalentProperty": {
        "@id": "http://purl.org/ontology/bibo/pageStart"
      },
      "rdfs:comment": "The page on which the work starts; for example \"135\" or \"xiii\".",
      "rdfs:label": "pageStart",
      "schema:domainIncludes": [
        {
          "@id": "schema:PublicationIssue"
        },
        {
          "@id": "schema:Chapter"
        },
        {
          "@id": "schema:Article"
        },
        {
          "@id": "schema:PublicationVolume"
        }
      ],
      "schema:rangeIncludes": [
        {
          "@id": "schema:Integer"
        },
        {
          "@id": "schema:Text"
        }
      ],
      "schema:sameAs": {
        "@id": "https://schema.org/pageStart"
      }
    }
```

**Comment by ljgarcia** [comment]
> > Regarding the rank and the taxonRank, maybe you could use the property hasCategoryCode, see http://bioschemas.org/devTypes/BioChemEntity/. In your case, the CategoryCode would have the codeValue pointing to what you currently have under rank, the inCodeset would point to the vocabulary where the rank term comes, and you still can use name for the taxonRank. Would that work for you?
> 
> I had not seen that but that's interesting indeed. However, I'd suggest to keep this for a later discussion with other biodiversity people. Is it ok with you @ljgarcia?

That is ok, yes

**Comment by frmichel** [comment]
Dearf @ljgarcia, @AlasdairGray,

I just committed a new version with the fixes except for adding the namespace in the properties names (tc, dwc) since you seem to disagree on this. Let me know it this is ok this way.

Franck.

**Comment by AlasdairGray** [comment]
> > Examples don't correspond with the property e.g. name has three examples only one of which is the name,
> 
> Right, my intend was to show the usage in context, exemplifying the relationship between name, scientificName and sicnetificNameAuthorhip. Fixed.

Examples showing complete usage should be provided in the [Examples](https://github.com/BioSchemas/specifications/tree/master/Taxon/examples) directory on github

**Comment by AlasdairGray** [comment]
There still seems to be an issue with `scientificName`. On the draft this is shown to come from schema.org but is actually coming from `dwc`

**Comment by AlasdairGray** [comment]
> @AlasdairGray all those prefixes will go away with the context. The original property source will only be visible inside the jsonld automatically created fromt the YAML. Properties from external ontologies are also used in schema.org, see for instance ScholarlyArticle where pageStart is equivalent to bibo:pageStart, that is not visible on the html but on the jsonld

I think it is helpful if we have some way of indicating where external terms have come from. Ideally we should also give their definition. Perhaps this needs to be done in the `Description` column.

**Comment by ljgarcia** [comment]
> > @AlasdairGray all those prefixes will go away with the context. The original property source will only be visible inside the jsonld automatically created fromt the YAML. Properties from external ontologies are also used in schema.org, see for instance ScholarlyArticle where pageStart is equivalent to bibo:pageStart, that is not visible on the html but on the jsonld
> 
> I think it is helpful if we have some way of indicating where external terms have come from. Ideally we should also give their definition. Perhaps this needs to be done in the `Description` column.

@AlasdairGray 
Should we better create an issue for that? This should not hold the Taxon spec as it goes beyond its scope.

**Comment by AlasdairGray** [comment]
@frmichel for clarity, I think that the only outstanding thing is around `scientificName`

> There still seems to be an issue with scientificName. On the draft this is shown to come from schema.org but is actually coming from dwc

**Comment by frmichel** [comment]
> There still seems to be an issue with `scientificName`. On the draft this is shown to come from schema.org but is actually coming from `dwc`

I'm sorry @AlasdairGray I'm probably missing something. Could you please point out the problem?

In the [spreadsheet](https://docs.google.com/spreadsheets/d/1hHE7SHz9qt6Eg6j6DQgI9BwhrlGOwkO68fzek78m62I/edit#gid=1261485211) the "type url" column gives "http://rs.tdwg.org/dwc/terms/scientificName", and in the property column it is an hyperlink with the same URI.

Same thing in the [YAML file](https://github.com/BioSchemas/bioschemas.github.io/blob/frmichel-patch-1/_devSpecs/Taxon.html) that this PR is about:

```
- property: scientificName
  expected_types:
  - Text
  description: ""
  type: ""
  type_url: http://rs.tdwg.org/dwc/terms/scientificName
  bsc_description: Full scientific name, with authorship and date information if known
  marginality: Optional
  cardinality: ONE
  controlled_vocab: ""
  example: |-
    {
        "dwc:scientificName": "Delphinapterus leucas (Pallas, 1776)"
}
```

**Comment by frmichel** [comment]
> I would expect scientificName to be black (as scientificNameAuthorship) not red as it is not a schema.org term.

Ok got it, sorry about that @AlasdairGray, thx for fixing it.

**Comment by frmichel** [comment]
Hi @AlasdairGray, @ricardoaat, I see that the [web page](http://bioschemas.org/devSpecs/Taxon/) has been updated, that's cool. I just noticed that I forgot to update the draft version to v. 0.2-draft in the yaml file. As a result the page still shows version v. 0.1-draft.

Could this be updated manually? I don't dare submit another PR just for that ;)... Let me know.

**Comment by AlasdairGray** [comment]
That's a good call. I've made the required change.

Hopefully PRs won't be so painful next time.

**Comment by frmichel** [comment]
Great, thx @AlasdairGray.

Yes I hope too, sorry for all the trouble.

Franck.

**Comment by ricardoaat** [comment]
I have also updated the [mapping file](https://docs.google.com/spreadsheets/d/1hHE7SHz9qt6Eg6j6DQgI9BwhrlGOwkO68fzek78m62I/edit#gid=1483018794), its always better to keep it updated with the information on the website even if its an small change done manually

**Comment by AlasdairGray** [comment]
@ricardoaat good call. I've just updated the type column for the scientificName property to reflect the other manual edit I made

**Pull request merged**